src/api/app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,84 @@
+# TODO: Please overwrite this comment with something explaining the controller target
+class AnnouncementsController < ApplicationController
+  before_action :set_announcement, only: [:show, :update, :destroy]
+  # Pundit authorization policies control
+  after_action :verify_authorized, except: :index
+  after_action :verify_policy_scoped, only: :index
+
+  # GET /announcements
+  def index
+    @announcements = policy_scope(Announcement.all)
+    xml = if @announcements.exists?
+            @announcements.to_xml(Announcement::DEFAULT_RENDER_PARAMS)
+          else
+            '<announcements></announcements>'
+          end
+    render xml: xml
+  end
+
+  # GET /announcements/1
+  def show
+    if @announcement.present?
```

Comment:
I think this is not needed because `set_announcement` will raise an exception if don't find the object.

Reply:
Then `set_announcement` needs to `render_error`

Reply:
yep

Reply:
not necessarily, you can make it like:

```
rescue_from MyException, with: :render_error
```
https://apidock.com/rails/ActiveSupport/Rescuable/ClassMethods/rescue_from

Reply:
> vpereira 10 minutes ago
> not necessarily

Yep, there is a generic handler in the application controller which result in something like this:

```ruby
<status code="not_found">
  <summary>Couldn't find Announcement with 'id'=324</summary>
</status>
```

I added a test to cover this as well.

Reply:
nice!